org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2020 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.java.resource;
+
+import java.util.List;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.util.JavaVersion;
+import org.eclipse.xtext.xbase.lib.Pure;
+
+public class JavaConfig {
+	public static class JavaConfigAdapter extends AdapterImpl {
+		private JavaConfig element;
+
+		public JavaConfigAdapter(JavaConfig element) {
+			this.element = element;
+		}
+
+		public JavaConfig get() {
+			return this.element;
+		}
+
+		@Override
+		public boolean isAdapterForType(final Object object) {
+			return object == JavaConfig.class;
+		}
+	}
+
+	private JavaVersion javaSourceLevel;
+
+	private JavaVersion javaTargetLevel;
+
+	public static JavaConfig findInEmfObject(Notifier emfObject) {
+		return (JavaConfig) EcoreUtil.getExistingAdapter(emfObject, JavaConfig.class);
```

Comment:
Must be `((JavaConfigAdapter)getExistingAdapter(emfObject, JavaConfig.class)).get())`